### PR TITLE
Fix crash with untuple and unresolved lambda

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fix crash with untuple and unresolved lambda

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 3d68efa99f787fa00fac2bbd68541d9b4e88882e with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63131/3d68efa99f787fa00fac2bbd68541d9b4e88882e/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by alexey-milovidov** [comment]
Why not throw an exception?

File: src/Analyzer/LambdaNode.cpp [hunk 1]
```diff
@@ -46,6 +48,11 @@ void LambdaNode::dumpTreeImpl(WriteBuffer & buffer, FormatState & format_state,
     getExpression()->dumpTreeImpl(buffer, format_state, indent + 4);
 }
 
+DataTypePtr LambdaNode::getResultType() const
+{
+    return result_type ? result_type : std::make_shared<DataTypeNothing>();
```
Comment (novikd) [inline src/Analyzer/LambdaNode.cpp:53]: It's better to correctly handle `untuple` case in `QueryAnalyzer::resolveFunction`.

**Review by novikd — CHANGES_REQUESTED** [review]

File: src/Analyzer/LambdaNode.cpp (hunk 1 above)
Comment (Algunenano) [reply src/Analyzer/LambdaNode.cpp:53]: You're right. I thought there would be more special cases, but it's only untuple and grouping AFAICS, and grouping works fine, so I'll handle it for `untuple` directly instead.

**Review by novikd — APPROVED** [review]

**Pull request merged**